### DISCUSSION TITLE
libinput-gestures failed to start as a desktop application.

### DISCUSSION THREAD

**Discussion by jackbluecz** [bulletmark/libinput-gestures#347] [Q&A]
I did the step by step correctly but when I start the program as the "readme" says, it returns the text: libinput-gestures failed to start as a desktop application.

I tried installing flathub gestures too, and this one worked, but returns: unable to read config file; please check permissions

**Comment by bulletmark** [comment]
Please note the **bolded sentence** in the opening paragraph of the troubleshooting section.

**Reply by jackbluecz** [reply]
sorry:
libinput-gestures: session xfce+x11 on Linux-5.10.0-21-amd64-x86_64-with-glibc2.31, python 3.9.2, libinput 1.16.4
Hash: 1cfe6ad528f2202a483fea24b3cd7f47
Traceback (most recent call last):
  File "/usr/bin/libinput-gestures", line 839, in <module>
    main()
  File "/usr/bin/libinput-gestures", line 718, in main
    confname = read_conf(args.conffile, CONFNAME)
  File "/usr/bin/libinput-gestures", line 637, in read_conf
    get_conf(confpath, confname)
  File "/usr/bin/libinput-gestures", line 597, in get_conf
    with conffile.open() as fp:
  File "/usr/lib/python3.9/pathlib.py", line 1241, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
IsADirectoryError: [Errno 21] Is a directory: '/home/diegovdds/.config/libinput-gestures.conf'

**Reply by jackbluecz** [reply]
ok, I apparently managed to resolve error 21 by replacing an empty folder that didn't have .config with the file that was in the git repository, now the output of "-l" goes like this:
libinput-gestures: session xfce+x11 on Linux-5.10.0-21-amd64-x86_64-with-glibc2.31, python 3.9.2, libinput 1.16.4
Hash: 1cfe6ad528f2202a483fea24b3cd7f47
Gestures configured in ~/.config/libinput-gestures.conf:
swipe up           _internal ws_up
swipe down         _internal ws_down
swipe left         xdotool key alt+Right
swipe right        xdotool key alt+Left
pinch in           xdotool key super+s
pinch out          xdotool key super+s
Could not determine touchpad device.

and the output of libinput-gestures-setup start :
libinput-gestures failed to start as a desktop application.

thank you for your attention

**Comment by bulletmark** [comment]
You have created this bug titled "failed to start as a desktop application" but surely you noticed the obvious error prior to that? It says "Could not determine touchpad device". That is your problem and why it is not starting.

Please paste the output of `libinput list-devices` here.

BTW, if you are only wanting the default configuration then you do not have to create any personal configuration file. `rm ~/.config/libinput-gestures.conf` unless you are planning to customize it.

**Reply by jackbluecz** [reply]
oh, sorry i'm resouved the problem, i just removed all and reinstall, thanks
